### DISCUSSION TITLE
Is that a good idea create a section pattern using custom block rather than block pattern?

### DISCUSSION THREAD

**Discussion by Tunglin-Li** [WordPress/gutenberg#54600] [Q&A]
I've recently developed a new website using a block theme for my clients. However, some of my clients are not very proficient with computers, which has made it challenging for them to grasp the intricacies of using the core block editor. Specifically, they find it difficult to work with the layout blocks, often clicking around without success in their quest to locate the appropriate group block.

One of the advantages of utilizing custom blocks is that it allows me to tailor the user interface to my clients' needs. Nevertheless, I have concerns about potential changes to block components in the future. For instance, if something as fundamental as RichText were to undergo a name or attribute change, it would necessitate extensive adjustments across my entire website, consuming a significant amount of time.

I'm aware of the templateLock: contentOnly function, but it has limitations and cannot accommodate substantial changes, such as altering YouTube embed URLs. Consequently, I'm at a bit of a standstill and uncertain about the best course of action. While I genuinely appreciate the flexibility and features of the block theme, I'm hesitant to revert to the classic theme with custom fields.

Has anyone else encountered a similar situation and can offer some guidance based on their experiences?

**Comment by tresorama** [comment]
I don't have what you are looking for, but in case you didn't know:
- **Block Patterns** are only text that is copy-pasted in the editor, there is no concept of "instance of a class/something".When they are inserted they "disappear" and the result is the same as if you did it all manually from scratch.
- **Block Variation** are equal to Block Pattern but they appears in the Block Inserter > Block (Tab) , as opposed to Block Pattern that appear in Block Inserter > Pattern (Tab), so a client would prefer Block Variation
- If you want to have something similar to a React/Vue/Svelte component use **Dynamic blocks**.
- **Dynamic blocks** are saved to Post Content as JSON data only (no HTML) and they are rendered to HTML when the page is requested because a visitor visit the public website. They are the opposite of Static blocks, which are saved to Post COntent as HTML, this HTML is rendered at the time of creating it (so when using the Block Editor).
`core/paragraph`, `core/heading` are static blocks.
`core/post-title` is a dynamic block.
If you use ACF Blocks (third party) they are dynamic blocks.
If your custom block can change in HTML markup (you will add a div, reorder HTML elements, or update CSS classes) you better use dynamic blocks, because you can change that in code and every instance of your block will show your changes without need to reopen the Block Editor. But if you "rename" an existing attribut of your block i think (I have never tested) you must re open the Block Editor in every instance and update content.

---
I also suggest you write here a concrete example of what kind of "workflow" you would like to have so anyone with more experience can better help you and the core devs can take your "struggle" as inspiration for the future...
